### PR TITLE
feat: kafka library changed 

### PR DESCRIPTION
**Pull request by Vaishnavi190900**
Moving kafka implementation from confluentic-kafka to segmentio-kafka.

### CODE REVIEW

**Review by sbchaos — APPROVED** [review]

File: changeEventProducer/kafka/writer.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	kafkaMessage := kafka.Message{
+		Topic: topic,
+		Value: messageBytes,
+	}
+	return w.send(kafkaMessage)
+}
+
+func (w *Writer) send(message kafka.Message) error {
+	err := w.kafkaWriter.WriteMessages(context.Background(), message)
```
Comment (gdgagan696) [inline changeEventProducer/kafka/writer.go:51]: will it take care of retries on its own?

File: core/schema/service.go [hunk 2]
```diff
@@ -191,9 +191,9 @@ func (s *Service) identifySchemaChange(ctx context.Context, request *changedetec
 		return fmt.Errorf("unable to insert event for namesapce %s , schema %s and version %d in DB, got error: %s", request.NamespaceID, request.SchemaName, request.Version, err.Error())
 	}
 	schemaChangeTopic := s.config.SchemaChange.KafkaTopic
-	retryInterval := time.Duration(s.config.KafkaProducer.RetryInterval) * time.Millisecond
-	if err := s.producer.PushMessagesWithRetries(schemaChangeTopic, sce, s.config.KafkaProducer.Retries, retryInterval); err != nil {
-		return fmt.Errorf("unable to push message to Kafka topic %s for schema change event %s: %s", schemaChangeTopic, sce, err.Error())
+	if err := s.producer.Write(schemaChangeTopic, sce); err != nil {
+		log.Printf("unable to push message to Kafka topic %s for schema change event %s: %s", schemaChangeTopic, sce, err.Error())
+		return err
```
Comment (gdgagan696) [inline core/schema/service.go]: can we use fmt.Errorf("unable to push message to Kafka topic %s for schema change event %s: %s", schemaChangeTopic, sce, err.Error())

File: changeEventProducer/kafka/writer.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
```
Comment (gdgagan696) [inline changeEventProducer/kafka/writer.go:5]: please remove kafka configs which are not being used.

**Review by gdgagan696 — CHANGES_REQUESTED** [review]

File: changeEventProducer/kafka/writer.go (hunk 1 above)
Comment (Vaishnavi190900) [reply changeEventProducer/kafka/writer.go:51]: yes, while creating Kafka writer, there is a config called MaxAttempts, which will take care of the retries.

**Review by gdgagan696 — APPROVED** [review]

**Pull request merged**